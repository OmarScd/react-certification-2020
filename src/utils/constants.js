const USERNAME_STORAGE_KEY = 'osa-reactcert-2020-username';

export { USERNAME_STORAGE_KEY };
