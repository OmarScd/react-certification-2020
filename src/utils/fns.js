// TODO: convert to fetch
const getQueryURL = (query) =>
  `https://content.googleapis.com/youtube/v3/search?part=id&part=snippet&q=${query}&maxResults=25&key=${process.env.REACT_APP_YOUTUBE_API_KEY}`;

const getRelatedURL = (videoId) =>
  `https://www.googleapis.com/youtube/v3/search?part=snippet&part=id&relatedToVideoId=${videoId}&type=video&maxResults=25&key=${process.env.REACT_APP_YOUTUBE_API_KEY}`;

export { getQueryURL, getRelatedURL };
