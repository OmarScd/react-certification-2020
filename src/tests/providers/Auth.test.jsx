import React from 'react';
import { render, cleanup } from '@testing-library/react';
import AuthProvider from '../../providers/Auth';

import '../../utils/storage';

jest.mock('../../utils/storage', () => {
  return {
    storage: {
      get: jest.fn(),
      set: jest.fn(),
      remove: jest.fn(),
    },
  };
});

describe('AuthProvider', () => {
  afterAll(cleanup);

  it('Renders correctly', () => {
    const { container } = render(
      <AuthProvider>
        <p>Mock content</p>
      </AuthProvider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
