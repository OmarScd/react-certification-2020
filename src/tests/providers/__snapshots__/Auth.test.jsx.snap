// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthProvider Renders correctly 1`] = `
<p>
  Mock content
</p>
`;
